magda-typescript-common/src/registry/AuthorizedRegistryClient.ts

```diff
@@ -14,27 +14,41 @@ import buildJwt from "../session/buildJwt";
 import { IncomingMessage } from "http";
 import { Maybe } from "tsmonad";
 
-export interface AuthorizedRegistryOptions extends RegistryOptions {
+export interface AuthorizedRegistryUserIdAndJwtSecretOptions
+    extends RegistryOptions {
+    userId: string;
     jwtSecret: string;
-    userId?: string;
+    jwt?: never;
```

Comment:
Why we need a [never](https://www.typescriptlang.org/docs/handbook/basic-types.html#never) type here?

A simple  
```typescript
type RegistryOptions{
    userId?: string;
    jwtSecret?: string;
    jwt?: string;
}
```
will not do?

At the end of the day, you probably still have to check it at runtime like:
```typescript
if(!options.jwt && (!options.userId || !options.jwtSecret)) {
  throw Error("JWT secret or userId can not be null");
}
```

Reply:
`never` will ensure only one of them is defined.